### PR TITLE
feat(content-block-horizontal): name change and deprecation (React)

### PR DESCRIPTION
**Pull request by NSDrowned**
### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6653

### Description

Renamed `Content Group Horizontal` to `Content Block Horizontal` , deprecating the old component

### CODE REVIEW

File: packages/react/src/components/ContentBlockHorizontal/README.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+> The ContentBlockHorizontal block pattern contains a collection of
+> [ContentItemHorizontal](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/tree/master/packages/react/src/components/ContentItemHorizontal)
+> content items.
+
+> 💡 Check our
+> [CodeSandbox](https://githubbox.com/carbon-design-system/carbon-for-ibm-dotcom/tree/master/packages/react/examples/codesandbox/components/ContentItemHorizontal)
+> example implementation.
+
+[![Edit @carbon/ibmdotcom-web-components](https://codesandbox.io/static/img/play-codesandbox.svg)](https://githubbox.com/carbon-design-system/carbon-for-ibm-dotcom/tree/master/packages/react/examples/codesandbox/components/ContentItemHorizontal)
```
Comment (kennylam) [inline packages/react/src/components/ContentBlockHorizontal/README.stories.mdx:15]:
```suggestion
[![Edit @carbon/ibmdotcom-web-components](https://codesandbox.io/static/img/play-codesandbox.svg)](https://githubbox.com/carbon-design-system/carbon-for-ibm-dotcom/tree/master/packages/react/examples/codesandbox/components/ContentBlockHorizontal)
```

Comment (kennylam) [inline packages/react/src/components/ContentBlockHorizontal/README.stories.mdx:15]:
```suggestion
[![Edit @carbon/ibmdotcom-web-components](https://codesandbox.io/static/img/play-codesandbox.svg)](https://githubbox.com/carbon-design-system/carbon-for-ibm-dotcom/tree/master/packages/react/examples/codesandbox/components/ContentBlockHorizontal)
```

**Review by kennylam — COMMENTED** [review]
Looks good @NSDrowned! Just one change.

File: packages/react/src/components/ContentBlockHorizontal/README.stories.mdx (hunk 1 above)
Comment (NSDrowned) [reply packages/react/src/components/ContentBlockHorizontal/README.stories.mdx:15]: Nice catch, updated the readme 👍

**Review by ariellalgilmore — APPROVED** [review]
LGTM!

**Pull request merged**